pkg/webservice/handlers.go

```diff
@@ -339,3 +341,67 @@ func getClusterConfig(w http.ResponseWriter, r *http.Request) {
 		http.Error(w, err.Error(), http.StatusInternalServerError)
 	}
 }
+
+func updateConfig(w http.ResponseWriter, r *http.Request) {
+	lock.Lock()
+	defer lock.Unlock()
+	writeHeaders(w)
+	requestBytes, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+		buildUpdateResponse(false, err.Error(), w)
+		return
+	}
+	// validation is already called when loading the config
+	schedulerConf, err := configs.LoadSchedulerConfigFromByteArray(requestBytes)
+	if err != nil {
+		buildUpdateResponse(false, err.Error(), w)
+		return
+	}
+	oldConf, err := updateConfiguration(string(requestBytes))
+	if err != nil {
+		buildUpdateResponse(false, err.Error(), w)
+		return
+	}
+	err = gClusterInfo.UpdateSchedulerConfig(schedulerConf)
+	if err != nil {
+		// revert configmap changes
+		_, err := updateConfiguration(oldConf)
+		if err != nil {
+			log.Logger().Error("Configuration rollback failed")
+		}
+		buildUpdateResponse(false, err.Error(), w)
```

Comment:
these lines seem to be problematic.
line 368 has overwritten the value of `err`, and line 372 is out of the `err != nil` check.
that means if `err == nil`, line 372 will give an NPE error

Reply:
Good catch! I was just focusing to fix the lint issue :). Fixed it